src/core/annotation.js

```diff
@@ -900,6 +902,61 @@ class TextAnnotation extends Annotation {
   }
 }
 
+class FreeTextAnnotation extends Annotation {
+  constructor(parameters) {
+    super(parameters);
+
+    this.data.annotationType = AnnotationType.FREETEXT;
+    let dict = parameters.dict;
+    this.pdfManager = parameters.pdfManager;
+
+    this.data.textColor = createRgbColor(dict.getArray('TextColor'));
+    this.data.textStyle = dict.get('DS');
+    this.data.fontSize = dict.get('FontSize');
+    this.data.opacity = dict.get('CA');
+    this.data.defaultAppearance = dict.get('DA');
+
+    this._preparePopup(dict);
+  }
+
+  getOperatorList(evaluator, task, renderForms) {
+    if (renderForms || this.appearance) {
+      return super.getOperatorList(evaluator, task, renderForms);
+    }
+
+    let data = this.data;
+
+    let opList = new OperatorList();
+    let appearanceStream = new Stream(stringToBytes(data.defaultAppearance));
+    let resourcesFonts = this.pdfManager.pdfDocument.catalog.fontCache;
+
+    return evaluator.getOperatorList({
+      stream: appearanceStream,
+      task,
+      resources: resourcesFonts,
+      operatorList: opList,
+    }).then(() => {
+      let a = opList.argsArray;
+      let i;
+      for (i = 0; i < opList.fnArray.length; i++) {
+        let fn = opList.fnArray[i];
+        switch (fn | 0) {
+          case OPS.setFont:
+            data.fontRefName = a[i][0];
```

Comment:
You don't seem to be using this anywhere in the display layer, so we always fall back to the default font. I think you can implement this similar to https://github.com/mozilla/pdf.js/blob/ff2df9c5b64a700e06694ff3375cd54c10ccedf4/src/display/annotation_layer.js#L451-L490, which you might also need to extract in a separate function then.